### PR TITLE
Cluster-Autoscaler: skip nodes currently under deletion in scale down

### PR DESCRIPTION
**Pull request by mwielgus**
Currently we may try to delete the same node multiple times.

cc: @MaciekPytel @jszczepkowski @fgrzadkowski

### CODE REVIEW

File: cluster-autoscaler/core/scale_down.go [hunk 1]
```diff
@@ -107,6 +107,15 @@ func (sd *ScaleDown) UpdateUnneededNodes(
 
 	// Phase1 - look at the nodes utilization.
 	for _, node := range nodes {
+
+		// Skip nodes marked to be deleted, if they were marked recently.
+		// Old-time marked nodes are again eligible for deletion - something went wrong with them
+		// and they have not been deleted.
+		deleteTime, _ := deletetaint.GetToBeDeletedTime(node)
+		if deleteTime != nil && (timestamp.Sub(*deleteTime) < MaxCloudProviderNodeDeletionTime || timestamp.Sub(*deleteTime) < MaxCloudProviderNodeDeletionTime) {
```
Comment (MaciekPytel) [inline cluster-autoscaler/core/scale_down.go]: Aren't both parts of the or condition exactly the same?

File: cluster-autoscaler/core/scale_down.go [hunk 2]
```diff
@@ -107,6 +107,15 @@ func (sd *ScaleDown) UpdateUnneededNodes(
...
+			glog.V(1).Info("Skipping %s from delete considerations - the node is currently being deleted", node.Name)
+		}
```
Comment (MaciekPytel) [inline cluster-autoscaler/core/scale_down.go]: continue?

Comment (mwielgus) [inline cluster-autoscaler/core/scale_down.go]: Done.

File: cluster-autoscaler/core/scale_down.go (hunk 1 above)
Comment (mwielgus) [inline cluster-autoscaler/core/scale_down.go]: Done.

**Comment by MaciekPytel** [comment]
\lgtm

**Pull request merged**